mne/datasets/fetch.py

```diff
@@ -0,0 +1,244 @@
+# Authors: Adam Li <adam2392@gmail.com>
+#
+# License: BSD Style.
+
+import sys
+import os
+import os.path as op
+from distutils.version import LooseVersion
+from shutil import rmtree
+
+from .config import (_bst_license_text, RELEASES,
+                     TESTING_VERSIONED, MISC_VERSIONED, MNE_DATASETS)
+from .. import __version__ as mne_version
+from ..utils import (logger, warn, _safe_input, _soft_import)
+from .utils import _dataset_version, _do_path_update
+
+
+_FAKE_VERSION = None  # used for monkeypatching while testing versioning
+
+
+def fetch_dataset(dataset_params, processor=None, path=None,
+                  force_update=False, update_path=True, download=True,
+                  check_version=False, return_version=False, accept=False,
+                  auth=None):
+    """Fetch an MNE-compatible dataset.
+
+    Parameters
+    ----------
+    dataset_params : dict of dict
+        The dataset name and corresponding parameters to download the dataset.
+        The dataset parameters that contains the following keys:
+        ``archive_name``, ``url``, ``folder_name``, ``hash``,
+        ``config_key`` (optional). See Notes.
+    processor : None | "zip" | "tar" | instance of pooch.Unzip | instance of pooch.Untar
+        The processor to handle the downloaded file. If ``None`` (default),
+        the files are left as is. If ``'zip'``, or ``'tar'`` will use
+        our internally defined `pooch.Unzip` or `pooch.Untar`.
+    path : None | str
+        Location of where to look for the {name} dataset.
```

Comment:
are all these `{name}` and `{conf}` fields still appropriate?  Is this docstring still getting `.format()` called on it now that it's a dataset-agnostic function?

Reply:
Got rid of it.